### PR TITLE
Fix past debts bug

### PR DESCRIPTION
**Pull request by jmul-dev**
This PR introduces the following changes:

- Loans should enter the "past debts" section on basis of whether repaidValue >= expectedRepaidValue(termEndTimestamp), not repaidValue >= principalAmount
- Fix the wrong border color around the Cancel open order button.

### CODE REVIEW

File: src/modules/Dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -85,10 +87,17 @@ class Dashboard extends React.Component<Props, States> {
                     debtRegistryEntry,
                 )) as any;
                 const repaymentSchedule = await adapter.getRepaymentSchedule(debtRegistryEntry);
+                const termEndTimestamp = repaymentSchedule.length
+                    ? repaymentSchedule[repaymentSchedule.length - 1]
+                    : moment().unix();
```
Comment (chrismin) [inline src/modules/Dashboard/Dashboard.tsx]: I don't think we can assume that the termEndTimestamp is the current time if we don't have a repayment schedule; in what scenarios would we not expect a repaymentSchedule to exist?

I think we should instead throw an error if we don't receive a repaymentSchedule.

File: src/modules/Dashboard/Dashboard.tsx [hunk 2]
```diff
@@ -154,13 +163,18 @@ class Dashboard extends React.Component<Props, States> {
                 const dharmaDebtOrder = (await adapter.fromDebtRegistryEntry(
                     debtRegistryEntry,
                 )) as any;
+                const repaymentSchedule = await adapter.getRepaymentSchedule(debtRegistryEntry);
+                const termEndTimestamp = repaymentSchedule.length
+                    ? repaymentSchedule[repaymentSchedule.length - 1]
+                    : moment().unix();
```
Comment (chrismin) [inline src/modules/Dashboard/Dashboard.tsx]: same comment as above

File: src/modules/Dashboard/Debts/DebtOrderHistory/DebtOrderRow.tsx [hunk 3]
```diff
@@ -44,9 +44,12 @@ class DebtOrderRow extends React.Component<Props, State> {
             return;
         }
         const valueRepaid = await dharma.servicing.getValueRepaid(debtOrder.issuanceHash);
+        const termEndTimestamp = debtOrder.repaymentSchedule.length
+            ? debtOrder.repaymentSchedule[debtOrder.repaymentSchedule.length - 1]
+            : moment().unix();
```
Comment (chrismin) [inline src/modules/Dashboard/Debts/DebtOrderHistory/DebtOrderRow.tsx]: same comment as above

File: src/modules/Dashboard/Investments/InvestmentHistory/InvestmentRow.tsx [hunk 4]
```diff
@@ -44,9 +44,12 @@ class InvestmentRow extends React.Component<Props, State> {
             return;
         }
         const earnedAmount = await dharma.servicing.getValueRepaid(investment.issuanceHash);
+        const termEndTimestamp = investment.repaymentSchedule.length
+            ? investment.repaymentSchedule[investment.repaymentSchedule.length - 1]
+            : moment().unix();
```
Comment (chrismin) [inline src/modules/Dashboard/Investments/InvestmentHistory/InvestmentRow.tsx]: same comment as above

**Comment by graemecode** [comment]
@jmul-dev @chrismin  I added a function to dharma.js that will come out in the next version, which gets the total expected repayment amount. This can be used instead of principal amount, and the change will be minimal.

**Comment by jmul-dev** [comment]
This is fixed!

**Comment by chrismin** [comment]
lgtm!

**Pull request merged**